src/_guides/language/coming-from/swift-to-dart.md

```diff
@@ -0,0 +1,2982 @@
+
+## Doc comments
+
+Regular comments work the same in Dart as they do in Swift.
+Using a double backslash (`//`) comments out everything
+beyond the double slash for the rest of the line,
+and `/* ... */` blocks comments spanning multiple lines.
+
+In addition to regular comments,
+Dart also has [doc comments][] that work in tandem
+with [dartdoc][]: a first-party tool that generates
+HTML documentation for Dart packages.
+It's considered best practice to place doc comments
+above all declarations for public members.
+You might notice that this process is similar to how
+you add comments for various documentation generation
+tools in Swift.
+
+[doc comments]: /guides/language/effective-dart/documentation
+[dartdoc]: /tools/dart-doc
+
+As in Swift, you define a doc comment by
+using three forward slashes instead of two (`///`):
+
+{:.include-lang}
+```dart
+/// The number of characters in this chunk when unsplit.
+int get length => ...
+```
+
+Surround types, parameter-, and method names
+with square brackets within doc comments. 
+
+{:.include-lang}
+```dart
+/// Returns the [int] multiplication result of [a] * [b].
+multiply(int a, int b) => a * b;
+```
+
+While there is support for JavaDoc-style doc comments,
+you should avoid them and use the `///` syntax. 
+
+{:.include-lang}
+```dart
+/** 
+ * The number of characters in this chunk when unsplit. 
+ * (AVOID USING THIS SYNTAX, USE /// INSTEAD.)
+ */
+int get length => ...
+```
+
+## Libraries and visibility
+
+Dart's visibility semantics are similar to Swift's,
+with Dart libraries being roughly equivalent to
+Swift modules.
+
+Dart offers two levels of access control:
+public and private. Methods and variables
+are public by default. Private variables
+are prefixed with the underline character (`_`),
+and are enforced by the Dart compiler.
+
+{:.include-lang}
+```dart
+final foo = 'this is a public property';
+final _foo = 'this is a private property';
+
+String bar() {
+  return 'this is a public method';
+}
+String _bar() {
+  return 'this is a private method';
+}
+
+// Public class
+class Foo {
+}
+
+// Private class
+class _Foo {
+},
+```
+
+Private methods and variables are scoped
+to their library in Dart, and to a module in Swift.
+In Dart, you can define a library in a file,
+while in Swift you must create a new build target
+for your module. This means that in a single Dart
+project you can define `n` libraries,
+but in Swift you must create `n` modules. 
+
+All files that are part of a library can gain access
+to all private objects in that library.
+But for security reasons, a file still needs
+to allow specific files to gain access to its
+private objects, otherwise any file&mdash;even from
+outside of your project&mdash;could register itself to your
+library and gain access to possibly sensitive data.
+In other words,
+private objects aren't shared across libraries.
+
+**animal.dart**
+{:.include-lang}
+```dart
+library animals;
```

Comment:
Consider adjusting this example to avoid specifying a named library and using a part file and instead using text to highlight they are in the same library of folder. Named libraries and parts are generally avoided now and not discussed elsewhere in this doc, so it may just cause unnecessary confusion.